### ISSUE TITLE
Fine_tune_BLIP2_on_an_image_captioning_dataset_PEFT.ipynb  loss is all NaN

### ISSUE THREAD

**Issue by z3ugma** [huggingface/notebooks#454]
peft/Fine_tune_BLIP2_on_an_image_captioning_dataset_PEFT.ipynb 
This notebook: 
https://github.com/huggingface/notebooks/blob/main/peft/Fine_tune_BLIP2_on_an_image_captioning_dataset_PEFT.ipynb 

Trains fine on Google Colab at https://colab.research.google.com/drive/16XbIysCzgpAld7Kd9-xz-23VPWmqdWmW?usp=sharing#scrollTo=upI97XEH6EKe 

using Python 3.10.12, Torch 2.1.0 

It does not train on my workstation - the loss collapses to NaN after just a few epochs: 

```
Loss: 6.078125
['a soccer player with his arms up in the air\n', 'mario balotelli celebrates after scoring against juventus\n']
Loss: 3.630859375
Loss: 4.01171875
Epoch: 2
Loss: 4.48046875
['cristiano ronaldo is the most expensive player in the world\n', 'a soccer player with his arms raised in celebration\n']
Loss: 3.25
Loss: 4.2734375
Epoch: 3
Loss: 4.0625
['a bald soccer player with a white shirt and blue shorts\n', "Juventus' Mario Mandzukic celebrates after scoring against Barcelona\n"]
Loss: 3.01953125
Loss: nan
Epoch: 4
Loss: nan
```

My workstation is also using Python 3.10.13, and Torch 2.1.0. What could be causing the loss to be all nan?

**Comment by z3ugma** [comment]
@younesbelkada you're the author of that sample notebook and the keeper of the football dataset on Hugging Face - any idea what might be causing the loss to go to nan?

**Comment by triangle959** [comment]
@z3ugma  the same problem. I started to get the nan loss in the 2nd batch of epoch 0. Have you solved it?

**Comment by triangle959** [comment]
I found an interesting thing. On Google Colab, the loss will not change to nan. It seems that there are still differences between Colab and the local notebook.

**Comment by jeffliu-LL** [comment]
I also ran into this issue recently with finetuning BLIP2, whereas it was working before. I haven't had a chance to pin it down, but it might be a package version issue with something introducing a breaking change?

**Comment by jeffliu-LL** [comment]
Rolling back to peft=0.5.0 was able to get the blip2 example working for me

**Comment by AntoniaSch** [comment]
@jeffliu-LL  which pytorch version are you using?

**Comment by z3ugma** [comment]
I will try rolling back to peft 0.5 with cuda 12.2 and Python 3.11. 

Will report back

**Comment by z3ugma** [comment]
No, still a problem: 

```
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2023 NVIDIA Corporation
Built on Tue_Aug_15_22:02:13_PDT_2023
Cuda compilation tools, release 12.2, V12.2.140
Build cuda_12.2.r12.2/compiler.33191640_0
Torch 2.1.2+cu121
Datasets 2.16.0
Python 3.11.7 | packaged by conda-forge | (main, Dec 15 2023, 08:38:37) [GCC 12.3.0]
PEFT 0.5.0
```

**Comment by z3ugma** [comment]
Still shows all `nan` after downgrading PyTorch  and PEFT unfortunately 

```
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2023 NVIDIA Corporation
Built on Tue_Aug_15_22:02:13_PDT_2023
Cuda compilation tools, release 12.2, V12.2.140
Build cuda_12.2.r12.2/compiler.33191640_0
Torch 2.0.1+cu117
Datasets 2.16.0
Python 3.11.7 | packaged by conda-forge | (main, Dec 15 2023, 08:38:37) [GCC 12.3.0]
PEFT 0.5.0
```

**Comment by z3ugma** [comment]
@jeffliu-LL will you put your versions of Python, PyTorch, and cuda from the working environment?

**Comment by z3ugma** [comment]
Here are the packages from the working Google Colab environment: 
```
Working: 
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2023 NVIDIA Corporation
Built on Tue_Aug_15_22:02:13_PDT_2023
Cuda compilation tools, release 12.2, V12.2.140
Build cuda_12.2.r12.2/compiler.33191640_0
Torch 2.0.1+cu117
Datasets 2.16.0
Transformers 4.35.2
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
PEFT 0.5.0
```

**Comment by z3ugma** [comment]
Still not working on Python 3.10 
some other version details of another nonworking version 
```
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2023 NVIDIA Corporation
Built on Tue_Aug_15_22:02:13_PDT_2023
Cuda compilation tools, release 12.2, V12.2.140
Build cuda_12.2.r12.2/compiler.33191640_0
Torch 2.0.1+cu117
Datasets 2.16.0
Transformers 4.35.2
Python 3.10.13 | packaged by conda-forge | (main, Dec 23 2023, 15:36:39) [GCC 12.3.0]
PEFT 0.5.0
SciPy 1.11.4
Pillow 9.4.0
```

**Comment by bit-bcilab** [comment]
seems like we meet the same problem :(

OS: Windows 10
CUDA: 11.8
Python 3.9.18 (main, Sep 11 2023, 14:09:26) [MSC v.1916 64 bit (AMD64)] on win32

Torch 2.1.2+cu118
Datasets 2.16.1
Transformers 4.36.2
PEFT 0.7.1
bitsandbytes 0.41.0

![Uploading 1.PNG…]()

**Comment by bit-bcilab** [comment]
![1](https://github.com/huggingface/notebooks/assets/50723971/8363cf04-546b-44cf-b578-16cd7cebbe2b)

**Comment by z3ugma** [comment]
@wushandinghua no, I've not yet had success

**Comment by pribadihcr** [comment]
Any solutions? have same problem

**Comment by eddie221** [comment]
I changed the type from torch.float16 to torch.float32. This works for me.

**Comment by pribadihcr** [comment]
> I changed the type from torch.float16 to torch.float32. This works for me. Hope this will also work for you.

change this:     pixel_values = batch.pop("pixel_values").to(device, torch.float32)

still has same prokem

**Comment by shams2023** [comment]
> peft/Fine_tune_BLIP2_on_an_image_captioning_dataset_PEFT.ipynb 此笔记本： https://github.com/huggingface/notebooks/blob/main/peft/Fine_tune_BLIP2_on_an_image_captioning_dataset_PEFT.ipynb
> 
> 在 Google Colab 上训练效果良好：[https://colab.research.google.com/drive/16XbIysCzgpAld7Kd9-xz-23VPWmqdWmW ?usp=sharing#scrollTo=upI97XEH6EKe](https://colab.research.google.com/drive/16XbIysCzgpAld7Kd9-xz-23VPWmqdWmW?usp=sharing#scrollTo=upI97XEH6EKe)
> 
> 使用Python 3.10.12、Torch 2.1.0
> 
> 它不在我的工作站上进行训练 - 损失在几个时期后就崩溃为 NaN：
> 
> ```
> Loss: 6.078125
> ['a soccer player with his arms up in the air\n', 'mario balotelli celebrates after scoring against juventus\n']
> Loss: 3.630859375
> Loss: 4.01171875
> Epoch: 2
> Loss: 4.48046875
> ['cristiano ronaldo is the most expensive player in the world\n', 'a soccer player with his arms raised in celebration\n']
> Loss: 3.25
> Loss: 4.2734375
> Epoch: 3
> Loss: 4.0625
> ['a bald soccer player with a white shirt and blue shorts\n', "Juventus' Mario Mandzukic celebrates after scoring against Barcelona\n"]
> Loss: 3.01953125
> Loss: nan
> Epoch: 4
> Loss: nan
> ```
> 
> 我的工作站也使用 Python 3.10.13 和 Torch 2.1.0。是什么导致损失全部为nan？



> Google Colab

Can you send me a copy of the code that deploys you locally? I also want to try it on my own computer instead of using Google Colab.
Thank you very much!

**Comment by shams2023** [comment]
> ![1](https://private-user-images.githubusercontent.com/50723971/295873833-8363cf04-546b-44cf-b578-16cd7cebbe2b.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAzMTg2OTQsIm5iZiI6MTcxMDMxODM5NCwicGF0aCI6Ii81MDcyMzk3MS8yOTU4NzM4MzMtODM2M2NmMDQtNTQ2Yi00NGNmLWI1NzgtMTZjZDdjZWJiZTJiLlBORz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzEzVDA4MjYzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYxOTE5MWFiMWNjMWNiNzBlMzQ0MzQ1ZjhhNzFmODZlNGFjNzg4NjBmNDgxZDIyYjdkZjkwNGY0ZWVhYWEyNDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Ff8qu57GEfbXsYkHdXTI4Np-U0t7sgQkIfpQcz2ayp8)

Can you send me a copy of the code that deploys you locally? I also want to try it on my own computer instead of using Google Colab.
Thank you very much!

**Comment by eddie221** [comment]
> > I changed the type from torch.float16 to torch.float32. This works for me. Hope this will also work for you.
> 
> change this: pixel_values = batch.pop("pixel_values").to(device, torch.float32)
> 
> still has same problem

Sorry for the late reply. 
I also change the model type from torch.float16 to torch.float32
There will need to be two modifications to the code:
1. model = Blip2ForConditionalGeneration.from_pretrained("ybelkada/blip2-opt-2.7b-fp16-sharded", device_map="auto", load_in_8bit=True, torch_dtype=torch.float32)
2. pixel_values = batch.pop("pixel_values").to(device, torch.float32)
Here is the notebook of my testing result:
[https://colab.research.google.com/drive/1j2jey-OqmtUa3IcI1kOcswWAWmiG4JKJ?usp=sharing](https://colab.research.google.com/drive/1j2jey-OqmtUa3IcI1kOcswWAWmiG4JKJ?usp=sharing)

**Comment by eddie221** [comment]
> > ![1](https://private-user-images.githubusercontent.com/50723971/295873833-8363cf04-546b-44cf-b578-16cd7cebbe2b.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAzMTg2OTQsIm5iZiI6MTcxMDMxODM5NCwicGF0aCI6Ii81MDcyMzk3MS8yOTU4NzM4MzMtODM2M2NmMDQtNTQ2Yi00NGNmLWI1NzgtMTZjZDdjZWJiZTJiLlBORz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzEzVDA4MjYzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYxOTE5MWFiMWNjMWNiNzBlMzQ0MzQ1ZjhhNzFmODZlNGFjNzg4NjBmNDgxZDIyYjdkZjkwNGY0ZWVhYWEyNDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Ff8qu57GEfbXsYkHdXTI4Np-U0t7sgQkIfpQcz2ayp8)
> 
> Can you send me a copy of the code that deploys you locally? I also want to try it on my own computer instead of using Google Colab. Thank you !

I test with the same code as the [https://github.com/huggingface/notebooks/blob/main/peft/Fine_tune_BLIP2_on_an_image_captioning_dataset_PEFT.ipynb](https://github.com/huggingface/notebooks/blob/main/peft/Fine_tune_BLIP2_on_an_image_captioning_dataset_PEFT.ipynb) on my computer.
